### ISSUE TITLE
Builds failing due Chrome

### ISSUE THREAD

**Issue by Pilzinsel64** [nextcloud-snap/nextcloud-snap#2478]
#### What dependency/component is out of date?
Looks like the link to Chrome is not available anymore. See the build errors:
https://github.com/nextcloud-snap/nextcloud-snap/actions/runs/5685667058/job/15411020535#step:8:82

This let fail all the builds at the moment.

Not sure where to get a new link for and how to change in the code. Anyone an idea?

**Comment by kyrofa** [comment]
Yeah, chrome broke all sorts of workflows with their latest shenanigans. We need to move to "chrome for testing". See [this blog post](https://developer.chrome.com/blog/chrome-for-testing/).

**Comment by Pilzinsel64** [comment]
I'm not sure, but looks like I'm a little bit lost here.
![grafik](https://github.com/nextcloud-snap/nextcloud-snap/assets/23138465/a4bbf58b-de8f-4f6d-aec1-47d6d465c563)

Feel free to take over. Branche is `fix/ci-chrome`. 😅

**Comment by faibz** [comment]
Just a guess, but I think the issue might be caused by the `webdrivers` gem. The [GitHub page for it](https://github.com/titusfortner/webdrivers#update-future-of-this-project) seems to mention Chrome breaking post 115, and that updating to Selenium 4.10 ([this?](https://rubygems.org/gems/selenium-webdriver/versions/4.10.0)) should work with Chrome 115. 

If that's correct, then perhaps the `webdrivers` entry should be dropped from `Gemfile.lock` and the `selenium-webdriver` version updated to 4.10.0.

**Comment by Pilzinsel64** [comment]
@faibz Good hint, thanks! Let's try it.
I made a commit just updating the version numbers a bit. Let's see if even this will work.

**Comment by Pilzinsel64** [comment]
Ok, I now tried to removing the `Gemfile.lock` and re-run `bundle install`.

Using the branche `fix/ci-webdirvers` just for testing the testing now. :D
If it success before the year ends, I'll create fresh branches & PRs for `develop`, `26` and `25`. ;)

**Comment by faibz** [comment]
I opened [this PR](https://github.com/nextcloud-snap/nextcloud-snap/pull/2479) which works locally, but is now out of date and has conflicts with the latest version of this fix/ci-chrome branch. Perhaps what's building now will work too as it seems to update the right things (capybara and selenium-webdriver).

**Comment by Pilzinsel64** [comment]
@faibz Thanks a lot!

Just in that minute I also transfered your changes to a fresh branch for a clean history. I also target Ruby 3.2.2. Let's see if one of our tests work in GitHub Actions now. :D

**Comment by faibz** [comment]
Looks like mine failed because a newer version of Ruby is required. Fingers crossed yours works :crossed_fingers:

**Comment by Pilzinsel64** [comment]
Hm, well, at least it starts testing. But it fails several points because the following:

```
TypeError:
        no implicit conversion of URI::HTTP into String
```

There is something to adjust to the workflows. 🤔

**Comment by Pilzinsel64** [comment]
I think I got it. Let's see if the change now works.

**Comment by kyrofa** [comment]
@faibz you're right, selenium >= 4.6 has Selenium Manager, which supports fetching chromedrivers, but I've learned more recently that this just isn't the way chrome is meant to be tested-- you should be manually installing Chrome for Testing, pinning the versions of both chrome and chromedriver.

However, it looks like chrome is coming from the github action worker, which we don't really have control over. As a result, we might as well go with your approach for now and see how much mileage we can get out of it. If we need to install our own chrome and chromedriver, we can do that later.

Thanks for your help!

**Comment by Pilzinsel64** [comment]
@faibz That would be amazing, thanks a lot. I hope the tests will success then.

I now tried a few things but now can't beat this error here:
```
An error occurred in an `after(:context)` hook.
Failure/Error: wait_for_nextcloud
TypeError:
  no implicit conversion of Hash into String
# ./integration/spec_helper.rb:199:in `initialize'
# ./integration/spec_helper.rb:199:in `open'
# ./integration/spec_helper.rb:199:in `block in wait_for_nextcloud'
# ./integration/spec_helper.rb:228:in `block in wait_for'
# ./integration/spec_helper.rb:227:in `wait_for'
# ./integration/spec_helper.rb:197:in `wait_for_nextcloud'
# ./integration/change_mode_spec.rb:6:in `block (2 levels) in <top (required)>'
```

So using an older version is probably the best we can no at the moment.

**Comment by faibz** [comment]
@kyrofa Ah, I see :sob:

Sounds good. Fingers crossed this solution ends up working, and then we can hopefully put this to bed for a while :joy: 

@Pilzinsel64 From what I can tell, that issue is caused by that `open` call which no longer gets routed to URI.open so instead calls Kernel.open which tries to open a file. I  think if you were to use `URI.open(uri, ...)`, it'd work. That's what I've done in my latest commit in #2481.

**Comment by faibz** [comment]
Fair enough. The build for #2481 succeeded, but if we want as few changes as possible, let's see if the build for #2480 passes, and then go with that approach.

**Comment by Pilzinsel64** [comment]
Hm. As the build already successes with ruby 3.2.2, I suggest to take #2481 then, because at some point the update need to be done. Why not now if it already success?
I have no knowledge about ruby, but can be ruby 3 that bad? 😳

**Comment by kyrofa** [comment]
Nah, I was trying to save @faibz some work. I agree, if #2481 is working, let's go with it.

**Comment by Pilzinsel64** [comment]
Fixed in #2481 #2483 #2484
